hw/top_earlgrey/ip/ast/doc/_index.md

```diff
@@ -0,0 +1,1006 @@
+interface functions. For further details about AST resets, see
+[<u>Resets</u>](#resets) section.
+
+Note: There are several reason for routing leaf clocks back into AST
+instead of using the root clocks directly
+
+  - > The leaf clocks may be divided down from the root clock and that
+    > frequency is used to drive the interface. For example,
+    > clk\_src\_io\_clk\_o is 96MHz, but comportable modules use either
+    > 48MHz or 24MHz.
+
+  - > The leaf clocks and root clocks have very different clock tree
+    > depths and may be difficult for timing closure if they interacted
+    > directly.
+
+  - > Decouple AST internal design from OpenTitan top-level interfaces
+    > clock and reset selection.
+
+# Register Access Interface
+
+AST registers can be accessed via TL-UL interface. These registers are
+used for test and calibration purposes and are not required for runtime
+operation. See the [<u>Interface Signals
+Table</u>](#interface-signals-table) for more details.
+
+# ADC
+
+AST contains an analog to digital converter that can be used to sample
+various input signals. For OpenTitan this will primarily be used for
+[<u>debug cable detection</u>](https://www.sparkfun.com/products/14746).
+To activate the ADC, the corresponding [<u>comportable
+module</u>](https://docs.google.com/document/d/1_GwCccYp0zFI8OT5so4832OMEXkC-I306tSsM3i92nA/edit?usp=sharing)
+must first select the channel to sample, and then activate the ADC
+through \`adc\_pd\_i\`. Once activated, the ADC will produce an output
+synchronous to the adc controller.
+
+## ADC Usage Flow
+
+1.  > Activate the ADC by negating ‘adc\_pd\_i’
+
+2.  > Wait 30 uS for the ADC to wake up.
+
+3.  > Select an analog channel to measure by setting the corresponding
+    > bit in ‘adc\_chnsel\_i’ bus. This triggers a measurement.
+
+4.  > Wait unitl ‘adc\_d\_val’ is set and read the result via
+    > ‘adc\_d\_o’
+
+5.  > Clear ‘adc\_chnsel\_i’ bus to 0. Note that adc\_chnsel must be
+    > cleared to 0 before a new channel is selected.
+
+6.  > Repeat steps 3-5 if more channels or more measurements are
+    > required
+
+7.  > Deactivate the ADC by setting ‘adc\_pd’ to save power.
+
+![](./media/image3.png)
+
+[<u>https://wavedrom.com/editor.html</u>](https://urldefense.proofpoint.com/v2/url?u=https-3A__wavedrom.com_editor.html&d=DwMFaQ&c=ue8mO8zgC4VZ4q_aNVKt8G9MC01UFDmisvMR1k-EoDM&r=ORLxyPY9MsmpJYCWs7a9VBneW9h5vR8MnW3aYvrOooI&m=ts63KFxLdp_aXcU3h1dbe_sIHcNCW8am1RhZxv1PNbA&s=MzZVoXpb6wDp_3iGVjovwJ1b-NW3uLacU2X-WhLfAZ4&e=)
+
+{signal: \[ {node: '.a..b........', phase:0.2},
+
+{name: 'adc\_pd\_i' , wave: '10|..|.....|....|..1'},
+
+{name: 'clk\_ast\_adc\_i', wave: 'p.|..|.....|....|...'},
+
+{name: 'adc\_chnsel\_i' , wave: '0.|.3|..04.|....|0..'},
+
+{name: 'adc\_d\_val\_o' , wave: '0.|..|.1.0.|.1..|.0.'},
+
+{name: 'adc\_d\_o' , wave: 'x.|..|.3.x.|.4..|.x.', data: \['ch0', 'ch1',
+'ch1'\]},
+
+\],edge: \[ 'a\<-\>b wakeup time', \] }
+

```

Comment:
sorry i just realized this information isn't there. But it might be good in the documentation to give a rough idea of the sample time for ADC (i assume this is okay since we also talk about the wakeup time). 

We don't anything super detailed, just a rough range I think would be good.

Reply:
OK